src/riak_kv_qry.erl

```diff
@@ -161,11 +165,17 @@ lookup_field_positions(Mod, FieldIdentifiers) ->
 %% and the general validation rules should pick that up.
 %% If there are too many values given for the fields it returns an error.
 %%
--spec xlate_insert_to_putdata([[riak_ql_ddl:data_value()]], [pos_integer()], tuple(undefined)) ->
+%% Additionally, we will convert Value in {varchar, Value} into an
+%% integer when that value is a timestamp. We would do it as we
+%% repackage list to tuple here, rather than separately as a
+%% post-processing step on Data, to avoid another round of
+%% list_to_tuple(tuple_to_list())
+-spec xlate_insert_to_putdata([[riak_ql_ddl:data_value()]], [pos_integer()], tuple(undefined),
+                              [riak_ql_ddl:simple_field_type()]) ->
                               {ok, [tuple()]} | {error, string()}.
-xlate_insert_to_putdata(Values, Positions, Empty) ->
+xlate_insert_to_putdata(Values, Positions, Empty, FieldTypes) ->
```

Comment:
I was a bit worried that was something I started using without precedent, but a quick search shows it's relatively common. IBM has used it, there's a company called xlate that does translation services, etc.